sections/docker/Caching-for-shorter-build-time.md

```diff
@@ -0,0 +1,120 @@
+# Leverage caching to reduce build times
+
+## One paragraph explainer
+
+Docker images are a combination of layers, each instruction in your Dockerfile creates a layer. The docker daemon can reuse those layers between builds if the instructions are identicals or in the case of a `COPY` or `ADD` files used are identical.
+It is important to layout your Dockerfile correctly to reduce the number of moving parts in your build, the less updated instructions should be at the top and the ones constantly changing (like app code) should be at the bottom. 
```

Comment:
Maybe 'The instruction that triggers the longest operation should be at the top to ensure it happens only when really necessary'

Reply:
I partially agree with that, given that sometimes the build code step is the longest one. We don't want to put it at the top in those cases. I'll try to mix and match